### DISCUSSION TITLE
Escape sequences in buffer work only in crossterm and termion, not termwiz

### DISCUSSION THREAD

**Discussion by benjajaja** [ratatui/ratatui#1425] [Discussion]
It is possible to just write any escape sequence into a buffer cell, and crossterm and termion will flush them to the terminal. This makes it possible for example to show images if supported.

However with termwiz, it looks like the escape character gets escaped. Is there a way to get an escape sequence through buffer / widgets to termwiz too? Or should this be approached differently, and if so, how?

Related to https://github.com/benjajaja/ratatui-image/issues/1

**Comment by orhun** [comment]
There is a fundamental difference between how `termwiz` and other backends (e.g. `crossterm`) work. For example, while `crossterm` prints the symbols as-is to the buffer:

https://github.com/ratatui-org/ratatui/blob/main/src/backend/crossterm.rs#L190

`termwiz` uses enum variants for each control sequence and also any additional sequence while printing text:

https://github.com/ratatui-org/ratatui/blob/main/src/backend/termwiz.rs#L178

From [`Change::Text`](https://docs.rs/termwiz/latest/termwiz/surface/change/enum.Change.html#variant.Text):

> Add printable text. Control characters are rendered inert by transforming them to space.

So it does not really allow printing control characters. If you want to print an image, you apparently need to use [`Change::Image`](https://docs.rs/termwiz/latest/termwiz/surface/change/enum.Change.html#variant.Image) which requires a change in the `TermwizBackend` implementation of Ratatui. We currently don't handle that.

P.S. Or maybe there is a way of handling [`termwiz::escape::Action::PrintString`](https://docs.rs/termwiz/latest/termwiz/escape/enum.Action.html#variant.PrintString) with `BufferedTerminal` which I haven't found yet.
